File: app/src/main/java/com/iogarage/ke/pennywise/PaymentAdapter.java
package com.iogarage.ke.pennywise;

import android.content.Context;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.iogarage.ke.pennywise.entities.Payment;
import com.iogarage.ke.pennywise.util.Util;

import org.greenrobot.eventbus.EventBus;

import java.util.List;

/**
 * Created by Joshua on 11/1/2014.
 */
public class PaymentAdapter extends RecyclerView.Adapter<PaymentAdapter.ViewHolder> {
    private List<Payment> mDataset;
    private Context mContext;

    public PaymentAdapter(List<Payment> dataset, Context context) {
        mDataset = dataset;
        mContext = context;
    }

    // Not use static
    public class ViewHolder extends RecyclerView.ViewHolder {
        public TextView mTrxDate;
        public TextView mAmount;
        public TextView mDesc;


        public ViewHolder(View itemView) {
            super(itemView);
            mTrxDate = itemView.findViewById(R.id.txtpaydate);
            mDesc = itemView.findViewById(R.id.txtdesc);
            mAmount = itemView.findViewById(R.id.txtamount);

        }
    }

    @Override
    public int getItemCount() {
        return mDataset.size();
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {

        final Payment payment = mDataset.get(position);

        if (payment.getDescription().equals(mContext.getString(R.string.penalty))) {
            holder.mAmount.setTextColor(ContextCompat.getColor(mContext, R.color.md_red_400));
        } else if (payment.getDescription().equals(mContext.getString(R.string.topup))) {
            holder.mAmount.setTextColor(ContextCompat.getColor(mContext, R.color.md_blue_600));
        } else
            holder.mAmount.setTextColor(ContextCompat.getColor(mContext, R.color.primary_dark));

        holder.mAmount.setText(Util.formatCurrency(payment.getAmountpaid()));
        holder.mTrxDate.setText(Util.formatDate(payment.getPaymentdate()));
        holder.mDesc.setText(TextUtils.isEmpty(payment.getDescription()) ? "Payment" : payment.getDescription());

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EventBus.getDefault().post(new PaymentSelected(payment));
            }
        });


    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(
                R.layout.payment_list_item, parent, false);
        ViewHolder holder = new ViewHolder(view);
        return holder;
    }
}

File: app/src/main/java/com/iogarage/ke/pennywise/views/payments/PaymentAdapter.kt
package com.iogarage.ke.pennywise.views.payments

import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.iogarage.ke.pennywise.R
import com.iogarage.ke.pennywise.domain.entity.Payment
import com.iogarage.ke.pennywise.util.asDateString
import com.iogarage.ke.pennywise.util.toCurrency

/**
 * Created by Joshua on 11/1/2014.
 */
class PaymentAdapter(
    private val mDataset: List<Payment>,
    private val onItemClicked: (Payment) -> Unit
) :
    RecyclerView.Adapter<PaymentAdapter.ViewHolder>() {
    // Not use static
    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mTrxDate: TextView
        var mAmount: TextView
        var mDesc: TextView

        init {
            mTrxDate = itemView.findViewById(R.id.txtpaydate)
            mDesc = itemView.findViewById(R.id.txtdesc)
            mAmount = itemView.findViewById(R.id.txtamount)
        }
    }

    override fun getItemCount(): Int {
        return mDataset.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val payment = mDataset[position]
        if (payment.description == holder.itemView.context.getString(R.string.penalty)) {
            holder.mAmount.setTextColor(
                ContextCompat.getColor(
                    holder.itemView.context,
                    R.color.md_red_400
                )
            )
        } else if (payment.description == holder.itemView.context.getString(R.string.topup)) {
            holder.mAmount.setTextColor(
                ContextCompat.getColor(
                    holder.itemView.context,
                    R.color.md_blue_600
                )
            )
        } else holder.mAmount.setTextColor(
            ContextCompat.getColor(
                holder.itemView.context,
                R.color.primary_dark
            )
        )
        holder.mAmount.text = payment.amountPaid.toCurrency()
        holder.mTrxDate.text = payment.paymentDate.asDateString()
        holder.mDesc.text =
            if (TextUtils.isEmpty(payment.description)) "Payment" else payment.description
        holder.itemView.setOnClickListener {
            onItemClicked(payment)
        }
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(
            R.layout.payment_list_item, parent, false
        )
        return ViewHolder(view)
    }
}